### PR TITLE
Update to Ember 4.12.0 LTS

### PR DESCRIPTION
**Pull request by Poltergeistz**
# Description

Upgrading frontend to Ember v4.12.0  + fixing minor deprecation warnings

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

# Related services

- N/A

# How to test 

1. Run the frontend on this branch.
2. run npm install
3. Test the app by using it.

Note : I already tested it and no issues so far.

# Links to other PR's

- N/A

# Notes

Bump frontend on `app-worship-decisions-database` after release

### CODE REVIEW

File: .woodpecker/.verify-pr.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 pipeline:
   install:
-    image: danlynn/ember-cli:4.4.0
+    image: danlynn/ember-cli:4.12.1
```
Comment (Windvis) [inline .woodpecker/.verify-pr.yml]: It's possible that this changed the node version (which might have issues with the lock file if it also brings in a newer npm). I would verify if the node in this image matches the one provided by madnificent/docker-ember (which we use in the Dockerfile). I think the madnificent one still uses node 16.

File: app/styles/app.css
Comment (Windvis) [inline app/styles/app.css:1]: I would delete this file and add "--allow-empty" or something similar (check the stylelint docs) to the npm script. In a future PR we can then [configure stylelint for sass](https://github.com/lblod/frontend-lpdc/commit/c9ad866d67ad66d4e6ec4a8307cd29b6ed19efec).

File: .woodpecker/.verify-pr.yml (hunk 1 above)
Comment (Windvis) [reply .woodpecker/.verify-pr.yml]: The ember version itself is not really important and don't need to match. Global ember-cli will always use the local project version when available.

File: package.json [hunk 2]
```diff
@@ -66,42 +73,44 @@
     "ember-cli-terser": "^4.0.2",
     "ember-composable-helpers": "^5.0.0",
     "ember-concurrency": "^2.2.1",
-    "ember-data": "~3.28.6",
-    "ember-export-application-global": "^2.0.1",
-    "ember-fetch": "^8.1.1",
+    "ember-data": "~4.11.3",
```
Comment (Windvis) [inline package.json]:
```suggestion
    "ember-data": "~4.12.3",
```

I think Ember Data 4.12 was released slightly to late, so they missed the blueprint bump..

**Review by Windvis — APPROVED** [review]
Haven't run the app, but other than some small things this looks good 👍.

File: .woodpecker/.verify-pr.yml (hunk 1 above)
Comment (Poltergeistz) [reply .woodpecker/.verify-pr.yml]: > It's possible that this changed the node version (which might have issues with the lock file if it also brings in a newer npm). I would verify if the node in this image matches the one provided by madnificent/docker-ember (which we use in the Dockerfile). I think the madnificent one still uses node 16.

Just checked, it uses node 18, will update the image

**Comment by Poltergeistz** [comment]
> Haven't run the app, but other than some small things this looks good 👍.

Thanks for the review :slightly_smiling_face:

**Comment by Windvis** [comment]
Thanks, It seems I forgot about this one. I gave it the last push. Merging 👍.

**Pull request merged**